### PR TITLE
GH227: Add documentation/sample for the use of optional parameters

### PR DESCRIPTION
**Pull request by kcamp**
Made a first pass on adding a note about supporting optional params + an example.
If it needs additional content or cleanup, let me know!

### CODE REVIEW

File: src/Cake.Web/App_Data/docs/fundamentals/aliases.md [hunk 1]
```diff
@@ -40,6 +40,17 @@ public static class MyCakeExtension
...
 
+    [CakeMethodAlias]
+    public static int GetMagicNumberOrDefault(this ICakeContext context, bool value, Func<int> defaultValueProvider = null)
+    {
+        if (value)
+        {
+            return int.MinValue;
+        }
+
+        return defaultValueProvider == null? int.MaxValue : defaultValueProvider();
```
Comment (gep13) [inline src/Cake.Web/App_Data/docs/fundamentals/aliases.md]: @kcamp are you missing a space between null and the ?

**Comment by gep13** [comment]
@cake-build/team aside from a small comment, this looks good to me, and I think the choice of the page makes sense.

Thoughts?

If everyone is happy, when we release 0.18.0, we can pull in this documentation too.

File: src/Cake.Web/App_Data/docs/fundamentals/aliases.md (hunk 1 above)
Comment (kcamp) [inline src/Cake.Web/App_Data/docs/fundamentals/aliases.md]: @gep13 that was intentional to match formatting of the existing example.  Based on your comment I cleaned up both examples and refreshed the commit.

Comment (gep13) [inline src/Cake.Web/App_Data/docs/fundamentals/aliases.md]: @kcamp gotcha! Thanks for changing this!

**Comment by devlead** [comment]
@kcamp  your changes have been merged, thanks for your contribution 👍

**Comment by gep13** [comment]
Relates to #227

**Pull request merged**